pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFormatSpec.java

```diff
@@ -16,13 +16,18 @@
 package com.linkedin.pinot.common.data;
 
 import java.util.Objects;
+import java.util.TimeZone;
 import java.util.concurrent.TimeUnit;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.apache.commons.lang3.EnumUtils;
 
 import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
 import com.linkedin.pinot.common.data.DateTimeFieldSpec.TimeFormat;
+import org.joda.time.DateTimeZone;
```

Comment:
Why do we need both joda and java TimeZones?

Reply:
TimeZone understands long form timezone notation (like America/Los_Angeles, Asia/Kolkata etc), short form (like CST, IST) as well as GMT offset (GMT+0530, GMT-07:00).
DateTimeZone understands only long form timezone notation (like America/Los_Angeles, Asia/Kolkata etc). But DateTimeZone can understand TimeZone.
The DateTimeFormatter we use, requires a DateTimeZone input. Hence we parse timezone using TimeZone, and give that to DateTimeZone, which enables us to support all formats.